fastqheat/__main__.py

```diff
@@ -20,6 +20,44 @@
 USABLE_CPUS_COUNT = len(os.sched_getaffinity(0))
 
 
+def get_fasterqdump_version() -> Optional[str]:
+    try:
+        result = subprocess.run(
+            ['fasterq-dump', '--version'], text=True, capture_output=True, check=True
+        )
+    except FileNotFoundError:
+        return None
+    except subprocess.CalledProcessError as e:
+        logging.error(e.stderr or e.stdout)
+        raise
+    else:
+        return result.stdout.strip()
+
+
+def get_pigz_version() -> Optional[str]:
+    try:
+        result = subprocess.run(['pigz', '--version'], text=True, capture_output=True, check=True)
+    except FileNotFoundError:
```

Comment:
Just asking: why two different exceptions?

Reply:
`subprocess.run` raises `FileNotFoundError` if it's unable to find the executable. The executable may return a non-zero exit code, in which case `subprocess.run` will raise `subprocess.CalledProcessError`.